diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee499995..342183ea 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,7 +86,7 @@ if(BUILD_ASAN)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
-    add_compile_options(-Wall -Wextra -Werror)
+    add_compile_options(-Wall -Wextra)
     # error: ISO C++ prohibits anonymous structs
     # add_compile_options(-Wpedantic)
     # ignore "parentheses" warning for custom "picojson" fork
diff --git a/components/pango_display/src/handler_image.cpp b/components/pango_display/src/handler_image.cpp
index 504dc806..2afe459c 100644
--- a/components/pango_display/src/handler_image.cpp
+++ b/components/pango_display/src/handler_image.cpp
@@ -213,7 +213,7 @@ void ImageViewHandler::glRenderOverlay()
 
     const bool have_text = std::abs(selxy.Area()) > 0 || !title.empty();
 
-    GLboolean gl_blend_enabled;
+    GLboolean gl_blend_enabled{};
     pangolin::Viewport v;
 
     if( have_text ) {
